DOC: Added examples to pd.Index.get_loc

topper-123 opened the pull request:
This pull request adds some examples to ``Index.get_loc`` (actually to ``Index._index_shared_docs['get_loc']``) and clarifies the return value.

Commit:
Added examples to pd.Index.get_loc

Commit:
Added examples to pd.Index.get_loc

pep8speaks commented:
Hello @topper-123! Thanks for submitting the PR.

 - In the file [`pandas/core/indexes/base.py`](https://github.com/pandas-dev/pandas/blob/0aa4ef8c89bb4cd130c151cd38c4e73864961b56/pandas/core/indexes/base.py), following are the PEP8 issues :

> [Line 2454:1](https://github.com/pandas-dev/pandas/blob/0aa4ef8c89bb4cd130c151cd38c4e73864961b56/pandas/core/indexes/base.py#L2454): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 2466:1](https://github.com/pandas-dev/pandas/blob/0aa4ef8c89bb4cd130c151cd38c4e73864961b56/pandas/core/indexes/base.py#L2466): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

jreback commented on pandas/core/indexes/base.py:
```diff
@@ -2450,7 +2450,19 @@ def _get_unique_index(self, dropna=False):
+        >>> unique_index = pd.Index(list('abc'))
+        >>> unique_index.get_loc('b')
+        1
+        >>> monotonic_index = pd.Index(list('abbc'))
```
can you add a blank line between cases

jreback commented:
lgtm. ping push / green.

topper-123 commented:
ping

topper-123 commented:
@jreback Is this ok? 

(More of a general question): I force-pushed the change, is that ok or is it considered "bad form"?

Commit:
use target_klass so class name is properly updated for subclasses

Commit:
use klass + add examples for CategoricalIndex, IntervalIndex and
MultiIndex.

jschendel committed:
TST: Enable tests in test_tools.py (#17405)

Enabled tests that currently aren't running.  Small fix to make sure all tests pass.  Verified that the raised messages match expectations for TestToDatetimeUnit::test_frame.

topper-123 committed:
TST: remove tests and docs for legacy (pre 0.12) hdf5 support (#17404)

jbrockmendel committed:
Tslib unused (#17402)

topper-123 committed:
DOC: Cleaned references to pandas <v0.12 in docs (#17375)

jbrockmendel committed:
Remove unused _day and _month attrs (#17431)

closes #17429

topper-123 committed:
DOC: Clean-up references to v12 to v14 (both included) (#17420)

s-weigand committed:
BUG: Plotting Timedelta on y-axis #16953 (#17430)

* implemented fix for GH issue #16953

* added tests for fix of issue #16953

* changed comments for git issue to pandas style GH#

* changed linelength in tests, so all lines are less than 80 characters

* added whatsnew entry

* swaped conversion and filtering of values, for plot to also work with object dtypes

* refomated code, so len(line) < 80

* changed whatsnew with timedelta and datetime dtypes

* added support for datetimetz and extended tests

* added reason to pytest.mark.xfail

jreback committed:
COMPAT: handle pyarrow deprecation of timestamps_to_ms in .from_pandas with pyarrow < 0.6.0 (#17447)

closes #17438

jreback commented:
@topper-123 force pushing to your own branch is just fine and in-fact necessary after rebasing.

jreback commented on pandas/core/indexes/base.py:
```diff
@@ -2450,10 +2450,24 @@ def _get_unique_index(self, dropna=False):
+
+        Examples
+        ---------
+        >>> unique_index = pd.%(target_klass)s(list('abc'))
```
hmm these are only valid for Index/CategoricalIndex. Do these examples show up for other indexes (IIR we are using this combined doc-string for all sub-classes). You'll have to do this as an additional ``@Appender`` to those classes only.

topper-123 replied:
I think I will do this all over again. Many of the ``.get_loc`` methods are subtlely different, so it's difficult to have a common doc-string.

Before I do that, could you please explain how ``IntervalIndex.get_loc`` gets its doc-string? From the code it seems it should have one, but it does....

jreback replied:
hmm, I think its inheriting it from Index.get_loc, instead this should use the appender method (guess it was omitted somehow).

topper-123 committed:
DOC/TST: Add examples to MultiIndex.get_level_values + related changes (#17414)

jbrockmendel committed:
Dont re-pin total_seconds as it is already implemented (#17432)

mroeschke committed:
BUG: Return local Timestamp.weekday_name attribute (#17354) (#17377)

toobaz committed:
BUG: intersection of decreasing RangeIndexes (#17374)

closes #17296

jbrockmendel committed:
Remove property that re-computed microsecond (#17331)

topper-123 committed:
cleaned references to pandas v0.15 and v0.16 in docs (#17442)

deniederhut committed:
BUG: revert collision warning (#17298)

mroeschke committed:
cdef out dtype for _Timestamp._get_field (#17457)

GuessWhoSamFoo committed:
DOC: Add Timestamp, Period, Timedelta, and Interval to api.rst (#17424)

majiang committed:
DOC: to_json (#17461)

jschendel committed:
BUG: Index._searchsorted_monotonic(..., side='right') returns the left side position for monotonic decreasing indexes (#17272)

mattip committed:
COMPAT: Pypy tweaks (#17351)

jbrockmendel committed:
Replace * imports with explicit imports; remove unused declared constants (#17470)

GuessWhoSamFoo committed:
Removed Timedelta.is_populated and fixed spelling errors (#17469)

jbrockmendel committed:
PERF: Implement get_freq_code in cython frequencies (#17422)

jbrockmendel committed:
Fix typo in setup.py introduced by 17422 (#17473)

jbrockmendel committed:
Follow up to #17422 (#17472)

gfyoung committed:
MAINT: calcurate --> calculate in _doctools.py

matthax committed:
BUG: Fix TypeError caused by GH13374 (#17465)

jbrockmendel committed:
Remove incorrect kwds from DateOffset tests (#17486)

jbrockmendel committed:
Remove pyx dependencies from setup (#17478)

Licht-T committed:
ENH: Add Styler.where (#17474)

rvernica committed:
Add file-like object to docs (#17492)

jbrockmendel committed:
Implement _is_utc in timezones (#17419)

jbrockmendel committed:
Follow-up to #17419 (#17497)

jreback committed:
DOC: fix parquet example to not use ns

mpenkov committed:
Prevent UnicodeDecodeError in pivot_table under Py2 (#17489)

tnir committed:
DEPR: Add warning for True for dropna of SeriesGroupBy.nth (#17493)

jreback committed:
COMPAT: Iteration should always yield a python scalar (#17491)

xref #10904
closes #13236
closes #13256
xref #14216

Giftlin committed:
DOC: grammatical mistake (#17511)

topper-123 committed:
removed versionadded <0.17 (#17504)

Giftlin committed:
DOC: grammatical mistakes (#17512)

jreback committed:
COMPAT: followup to #17491 (#17503)

jbrockmendel committed:
De-privatize timezone funcs (#17502)

jschendel committed:
Make *_range functions consistent (#17482)

kirkhansen committed:
TST: Made s3 related tests mock boto (#17388)

TomAugspurger committed:
PERF: Avoid values in Categorical.set_categories (#17515)

Mater:

```python
In [1]: import pandas as pd; import numpy as np

In [2]: arr = ['s%04d' % i for i in np.random.randint(0, 500000 // 10,
                                                      size=500000)];
s = pd.Series(arr).astype('category')

In [3]: %timeit s.cat.set_categories(s.cat.categories)
68.5 ms ± 846 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

HEAD:

```python
In [1]: import pandas as pd; import numpy as np

In [2]: arr = ['s%04d' % i for i in np.random.randint(0, 500000 // 10,
                                                      size=500000)]
s = pd.Series(arr).astype('category')

In [3]: %timeit s.cat.set_categories(s.cat.categories)
7.43 ms ± 110 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Closes https://github.com/pandas-dev/pandas/issues/17508

jbrockmendel committed:
remove period_helper from non-period reqs (#17531)

jbrockmendel committed:
Fix bug where offset.copy() != offset (#17452)

TomAugspurger committed:
PERF: Faster CategoricalIndex from categorical (#17513)

jbrockmendel committed:
Remove unnecessary iNaT checks from _Period properties (#17421)

jschendel committed:
CLN: Fix Spelling Errors (#17535)

jbrockmendel committed:
Cut/paste (most) remaining tz funcs to tslibs/timezones (#17526)

Giftlin committed:
DOC: Spelling + grammar in chainmap_impl.py (#17548)

bashtage committed:
BUG: Set index when reading Stata file (#17328)

Ensures index is set when requested during reading of a Stata dta file
Deprecates and renames index to index_col for API consistence

closes #16342

Commit:
merge conflict

topper-123 commented:
Hi @jreback. I've updated the examples for other index classes as well, but I've messed up the commits, lots of other people's commit are included also.

Should I just delete this branch and make a new pull request?

jreback commented:
@topper-123 generally just push (or force push) to the same branch is easiest, but since you did it ok
